stl/src/float_multi_prec.hpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// implements multi-precision floating point arithmetic for numerical algorithms
+
+// This must be as small as possible, because its contents are
+// injected into the msvcprt.lib and msvcprtd.lib import libraries.
+// Do not include or define anything else here.
+// In particular, basic_string must not be included here.
+
+#pragma once
+
+#if defined(_M_FP_FAST) || defined(__FAST_MATH__)
+#error float_multi_prec.hpp should not be compiled with /fp:fast or -ffast-math
+#endif // defined(_M_FP_FAST) || defined(__FAST_MATH__)
+
+#include <intrin.h>

```

Comment:
What intrinsics are you using here that aren't in `intrin0.h`? Including `intrin.h` isn't the end of the world here since this file is in `stl/src` but I'd like to still avoid it.

Reply:
x86/x64: `_mm_fmsub_sd`, `_mm_set_sd`, `_mm_store_sd`
arm64: `vfma_f64`, `vld1_f64`, `vst1_f64`, `vneg_f64`

MSVC and clang define SSE2/FMA3 intrinsics differently. I've attempted to make both of them compile without `intrin.h`, but [the result](https://gist.github.com/statementreply/f17bff9e5adadb097b680eb42bad557a) looks messy.

Reply:
Agreed it looks messy. @StephanTLavavej what do you think? We could move the needed intrinsics into `intrin0.h`.